narrative_llm_agent/agents/analyst.py

```diff
@@ -6,62 +6,138 @@
+        else:
+            raise KeyError("Missing environment variable OPENAI_API_KEY")
+
+    def __init_agent(self: "AnalystAgent") -> None:
+        @tool(
+            "KBase documentation retrieval tool",
+            args_schema=AnalystInput,
+            return_direct=True,
+        )
         def kbase_docs_retrieval_tool(input: str):
-            """This tool has the KBase documentation. Useful for when you need to answer questions about how to use Kbase applications. Input should be a fully formed question. """
-            persist_directory = "./vector_db_kbase_docs"
-            return self._create_doc_chain(persist_directory=persist_directory).invoke({"query": input})
-            
-        @tool("Kbase app catalog retrieval tool", args_schema = AnalystInput, return_direct=True)
+            """This tool has the KBase documentation. Useful for when you need to answer questions about how to use Kbase applications. Input should be a fully formed question."""
+            return self._create_doc_chain(persist_directory=self._docs_db_dir).invoke(
+                {"query": input}
+            )
+
+        @tool(
+            "KBase app catalog retrieval tool",
+            args_schema=AnalystInput,
+            return_direct=True,
+        )
         def kbase_appCatalog_retrieval_tool(input: str):
-            """This tool has the KBase app catalog. Useful for when you need to find apps available in KBase. 
-            All apps in the catalog also have name, version tooltip, categories and description to help you to decide which app to use. Input should be a fully formed question. """
-            persist_directory = "./vector_db_app_catalog"
-            return self._create_doc_chain(persist_directory=persist_directory).invoke({"query": input})
-            
+            """This tool has the KBase app catalog. Useful for when you need to find apps available in KBase.
+            All apps in the catalog also have name, version tooltip, categories and description to help you to decide which app to use. Input should be a fully formed question."""
+            return self._create_doc_chain(persist_directory=self._catalog_db_dir).invoke(
+                {"query": input}
+            )
+
         self.agent = Agent(
             role=self.role,
             goal=self.goal,
             backstory=self.backstory,
             verbose=True,
             allow_delegation=True,
             llm=self._llm,
-            tools=[kbase_appCatalog_retrieval_tool, kbase_docs_retrieval_tool]
+            tools=[kbase_appCatalog_retrieval_tool, kbase_docs_retrieval_tool],
+        )
+
+    def _create_doc_chain(self, persist_directory: str | Path):
+        # Embedding functions to use
+        embeddings = OpenAIEmbeddings(openai_api_key=self._openai_key)
+        # Use the persisted database
+        vectordb = Chroma(
+            persist_directory=str(persist_directory), embedding_function=embeddings
```

Comment:
Pathlib is nice, but still needs to get cast as a str for langchain to handle it. Which is silly as pathlib has been accepted as the main tool for a while...